wagtail_localize/tests/test_edit_translation.py

```diff
@@ -89,82 +92,84 @@
 
 class EditTranslationTestData(WagtailTestUtils):
     def setUp(self):
-        self.login()
-        self.user = get_user_model().objects.get()
-
-        # Convert the user into an editor
-        self.moderators_group = Group.objects.get(name="Moderators")
-        for permission in Permission.objects.filter(
-            content_type=ContentType.objects.get_for_model(TestSnippet)
-        ):
-            self.moderators_group.permissions.add(permission)
-        for permission in Permission.objects.filter(
-            content_type=ContentType.objects.get_for_model(NonTranslatableSnippet)
-        ):
-            self.moderators_group.permissions.add(permission)
-        self.user.is_superuser = False
-        self.user.groups.add(self.moderators_group)
-        self.user.save()
-
-        # Create page
-        self.snippet = TestSnippet.objects.create(field="Test snippet")
-        self.home_page = Page.objects.get(depth=2)
-        self.page = self.home_page.add_child(
-            instance=TestPage(
-                title="The title",
-                slug="test",
-                test_charfield="A char field",
-                test_textfield="A text field",
-                test_emailfield="email@example.com",
-                test_synchronized_emailfield="email@example.com",
-                test_slugfield="a-slug-field",
-                test_urlfield="https://www.example.com",
-                test_richtextfield=RICH_TEXT_DATA,
-                test_streamfield=StreamValue(
-                    TestPage.test_streamfield.field.stream_block,
-                    STREAM_DATA,
-                    is_lazy=True,
-                ),
-                test_snippet=self.snippet,
+        with patch.object(transaction, "on_commit", side_effect=lambda func: func()):
```

Comment:
I really don't think these setup tasks need to be wrapped like this.
we are only interested on the create/update translation tests to issue a on_commit

Reply:
Yeah that's a good point, moved the patch to an overriden setup method in the relevant test class. (I'm wrapping the setup task because the Translation.save_target method used executes my changed code)

Unless I'm missing something, the only way to bring the patch into the test_edit_page_translation method itself would be to split up the setup method and call the relevant part of it in the actual test